index.js

```diff
@@ -174,7 +174,16 @@ InfluxDB.prototype.getDatabaseNames = function (callback) {
     if (err) {
       return callback(err, results)
     }
-    return callback(err, _.map(results[0].series[0].values, function (dbarray) { return dbarray[0] }))
+
+    if (_.isArray(results) && !_.isEmpty(results) && _.isArray(results[0].series) &&
+        !_.isEmpty(results[0].series) && _.isArray(results[0].series[0].values)) {
+
+      return callback(err, _.map(results[0].series[0].values, function (dbarray) {
+        if (_.isArray(dbarray) && !_.isEmpty(dbarray)) {
+          return dbarray[0]
+        }
+      }))
+    } else { return callback(new Error('bad response from server', results))}
```

Comment:
I like it, much tidier than mine. 

I still think it needs more type checking as its still easy to crash with a crafted response(I know in normal circumstances this would prob not happen) but you cant catch it outside the lib so I think it should be handled here.

So how about something like this?

```javascript
// These still crash the lib
var results = [{
      series: [{values: 'nope'}]
}]

var results = [{
        series: [{values: [null, undefined]}]
}];
// --------------------------------------------

// More type checking solves above crashes

var names = _.get(results, '[0].series[0].values')
if (!_.isArray(names)) {
  return callback(new Error('bad response from server', results))
}

callback(null, names.map(function (dbarray) {
  if (_.isArray(dbarray)) {
    return dbarray[0]
  }
}))
```